tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -256,6 +257,28 @@ def test_circuit_wire_validation(self):
         with pytest.raises(qml.wires.WireError, match=r"Cannot run circuit\(s\) on"):
             dev.preprocess([circuit_valid_0, invalid_circuit])
 
+    @pytest.mark.parametrize(
+        "mp_fn,mp_cls,shots",
+        [
+            (qml.sample, SampleMP, 10),
+            (qml.state, StateMP, None),
+            (qml.probs, ProbabilityMP, None),
+        ],
+    )
+    def test_measurement_is_swapped_out(self, mp_fn, mp_cls, shots):
+        """Test that preprocessing swaps out any MP with no wires or obs"""
+        dev = DefaultQubit2(wires=3)
+        original_mp = mp_fn()
+        exp_z = qml.expval(qml.PauliZ(0))
+        qs = qml.tape.QuantumScript([qml.Hadamard(0)], [original_mp, exp_z], shots=shots)
+        tapes, _, _ = dev.preprocess(qs)
+        assert len(tapes) == 1
+        tape = tapes[0]
+        assert tape.operations == qs.operations
+        assert tape.measurements != qs.measurements
+        assert qml.equal(tape.measurements[0], mp_cls(wires=[0, 1, 2]))
```

Comment:
Could use `==` now 😄

Reply:
that's going to take some getting used to. but nice 😎